### DISCUSSION TITLE
Updating grammar bindings (N-API, etc)

### DISCUSSION THREAD

**Discussion by segevfiner** [tree-sitter/tree-sitter#3192] [General]
Since the latest release, we now need to update the language bindings for all grammars so they are usable with the new versions of node-tree-sitter, anything I can do to help? Will a script to automate the opening of a PR with the update help?

@amaanq @ObserverOfTime @verhovsky

**Comment by ObserverOfTime** [comment]
The automated part is just `tree-sitter generate`. Anything beyond that requires manual intervention.

**Comment by segevfiner** [comment]
Not exactly, we can do a script that does:
1. Clones the repo.
2. Run `tree-sitter generate`
3. Opens a PR.

And it's manual from that point. Can help doing it to many grammars at once a bit.

**Reply by verhovsky** [reply]
It should also check that the existing binding.cc file is a generated one and wasn't changed. For example, I've seen a grammar that exports multiple languages at once.
